node/lib/cmd/commit-shadow.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2017, Two Sigma Open Source
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of git-meta nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+"use strict";
+
+const co = require("co");
+
+/**
+ * this module is the entrypoint for the `commit-shadow` command.
+ */
+
+/**
+ * help text for the `commit-shadow` command
+ *
+ * @property {String}
+ */
+exports.helpText =
+    `Create a "shadow" commit, leaving index and HEAD unchanged.`;
+
+/**
+ * description of the `commit-shadow` comand
+ *
+ * @property {String}
+ */
+exports.description  = `Create a "shadow" commit containing all local
+modifications (including untracked files) to all sub-repos and print its SHA.
+If there are no local modifications, print the SHA of HEAD.  Do not modify the
+index or update HEAD to point to the created commit.  Note that this command
+ignores changes to the meta-repo.  Note that this command is meant for
```

Comment:
Wait, I thought it handled added/removed submodules?

Reply:
Sorry, I should clarify that it ignore non-submodule changes to the meta-repo. The reasons for this are (a) it's faster if we don't have to do the equivalent of a `git-status` in the meta-repo and (b) I have some work to do in my own `getRepoStatus` method for this case (where `ignoreIndex` is true) so that it's doesn't see files in new submodules as untracked.